### ISSUE TITLE
Bundle.module does not find resources after deployment

### ISSUE THREAD

**Issue by Peter-Schorn** [vapor-community/heroku-buildpack#63]
I would like to host the documentation I've created using Apple's DocC compiler on heroku. To do this, my server needs access to a doccarchive folder. I declared this folder as a resource in my `pacakge.swift` file as shown below:
```swift
.target(
    name: "VaporDocC",
    dependencies: [
        .product(name: "Vapor", package: "vapor"),
    ],
    resources: [
        .copy("SpotifyWebAPI.doccarchive")
    ]
),
```
When I run my app from Xcode, I am able to access this folder. However, when I deploy my app to heroku, it crashes because it cannot find this folder.

**How do I make resources accessible by my app on ~~vapor~~ Heroku?**

**Comment by vzsg** [comment]
Hi. Thanks for opening the issue, and sorry for the delayed response.
Unfortunately I have no idea how resources work on Linux, but I'll ask around.

**Comment by vzsg** [comment]
Also, do you have that "SpotifyWebAPI.doccarchive" folder committed to source control? The buildpack, especially on Swift 5.4 and before, will not compile DocC by itself.

**Comment by vzsg** [comment]
Do I assume correctly that you are trying to deploy this project? https://github.com/Peter-Schorn/SpotifyAPIDocs

**Comment by vzsg** [comment]
Doubling down on that "help-wanted" label, I deployed your repository to Heroku and found this in the logs. (It would've been nice to see in in the original issue).

```
2021-06-29T18:30:48.624997+00:00 heroku[web.1]: Starting process with command `Run serve --env production --hostname 0.0.0.0 --port 4728`
2021-06-29T18:30:52.295359+00:00 app[web.1]: VaporDocC/resource_bundle_accessor.swift:11: Fatal error: could not load resource bundle: from /tmp/codon/tmp/cache/heroku-20/swiftenv/versions/5.4.2/usr/bin/SpotifyAPIDocs_VaporDocC.resources or /tmp/build_a14d8f44/.build/x86_64-unknown-linux-gnu/release/SpotifyAPIDocs_VaporDocC.resources
2021-06-29T18:30:52.791165+00:00 app[web.1]: 0x7f3bbe6c03bf
2021-06-29T18:30:52.791183+00:00 app[web.1]: 0x7f3bbe843307
2021-06-29T18:30:52.791184+00:00 app[web.1]: 0x55dc9ded7ad7, closure #1 () -> Foundation.Bundle in variable initialization expression of static (extension in VaporDocC):Foundation.Bundle.module : Foundation.Bundle at /tmp/build_a14d8f44/.build/x86_64-unknown-linux-gnu/release/VaporDocC.build/DerivedSources/resource_bundle_accessor.swift:11
```

Unfortunately, this suggests that Swift hardcodes the build path to the binary, so even though the resources folder is copied right next to the binary (which is done by the buidpack), the app will not see them.

**Comment by Peter-Schorn** [comment]
> Do I assume correctly that you are trying to deploy this project? https://github.com/Peter-Schorn/SpotifyAPIDocs

Yes, I want to deploy this project.

> Unfortunately, this suggests that Swift hardcodes the build path to the binary, so even though the resources folder is copied right next to the binary (which is done by the buidpack), the app will not see them.

I assumed that the resources folder was not copied at all. Is there any way I can reliably retrieve the correct path to the resources folder? Perhaps I should create a docker image instead and deploy that. However, I'm not sure how to correctly set up that either.

**Comment by vzsg** [comment]
If you want a reliable solution, don't use `Bundle.module` until we can figure something out.

I have two workaround ideas in the meantime:
- Stick to serving files directly from the project folder (like FileMiddleware)
- Use `Bundle(path: "/app/.swift-bin/SpotifyAPIDocs_VaporDocC.resources")` as a fallback if Bundle.module returns nil.

**Comment by vzsg** [comment]
For the record, I found where the hardcoded temporary paths are coming from [in SPM's sources](https://github.com/apple/swift-package-manager/blob/main/Sources/Build/BuildPlan.swift#L668-L682).

I have absolutely no idea why it's implemented like that, and why the toolchain's bin folder end up being used as the main bundle path. Or how Swift apps with resources are supposed to be portable between systems...

**Comment by vzsg** [comment]
I've found a [bug report](https://bugs.swift.org/browse/SR-14555) about this exact issue, and it will be fixed in Swift 5.5.
The fix is already part of the 5.5-DEVELOPMENT-SNAPSHOT-2021-06-26-a version, which is supported by the buildpack.

So, for the third workaround, you can set the Swift version used to `5.5-DEVELOPMENT-SNAPSHOT-2021-06-26-a` with either a `.swift-version` file or with the `SWIFT_VERSION` config variable and build again.

Until then, unfortunately there is nothing I can do to fix Bundle.module.
